openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -775,13 +788,350 @@ function scale(IndicatorObj) {
+        for (var i = 0; i < allSVIThemes.length; i++) {
+            $('#webGlThematicSelection').append('<option class="3">'+allSVIThemes[i]+'</option>');
+        }
+    }
+
+    // Add primary indicators to selection menu
+    if (allPrimaryIndicators.length > 0) {
+        $('#webGlThematicSelection').append('<optgroup label="Primary Indicators">');
+        for (var i = 0; i < allPrimaryIndicators.length; i++) {
+            $('#webGlThematicSelection').append('<option class="4">'+allPrimaryIndicators[i]+'</option>');
+        }
+    }
+
+    $('#webGlThematicSelection').show();
+
+    // Manage the thematic map selection menu
+    // Set the map selection menu to the first multi group dropdown option
+    if (selectedIndicator === undefined) {
+        $("#webGlThematicSelection").val($("#webGlThematicSelection option:first").val());
+        mapboxGlLayerCreation();
+    } else {
+        $('#webGlThematicSelection').val(selectedIndicator);
+    }
+
+    // Execute the mapboxGlLayerCreation when there has been a weight change
+    if (weightChange > 0) {
+        mapboxGlLayerCreation(weightChange);
+    }
+
+    $('#webGlThematicSelection').change(function() {
+        mapboxGlLayerCreation();
+    });
+
+}
+
+function mapboxGlLayerCreation() {
+    // There are 4 cases for managing the state of the map:
+    // 1. The user has created a map for the first time
+    // 2. The user changes the map theme
+    // 3. The user changes the weights
+    // 4. The user changes the project
+
+    // Case 2:
+    // The data used for thematic map has not changed, so we
+    // keep the map source, destroy and recreate the map layers
+
+    // Case 3:
+    // The weight change will alter the underlying data, so we
+    // update the map style and source
+
+    // Case 3.1
+    // Only reload the map for the appropriate weight change level
+
+    // Case 4:
+    // A new project has been loaded into the application
+
+    var weightChange = 0;
+
+    if (arguments[0]) {
+        weightChange = arguments[0];
+    }
+
+    selectedIndicator = $('#webGlThematicSelection').val();
+
+    // Avoid unnecessarily redrawing the map.
+    // Only redraw the map when:
+        // 1) the level of the weight changes is lower than the level of the thematic map menu.
+            // Example 1: if the user is viewing the SVI thematic map, and then changes any theme, or
+            // primary indicator weight, the map must be refreshed.
+            // Example 2: If the user is viewing a theme thematic map (e.g. Economy), and then changes
+            // the SVI weight, the map should not be redrawn.
+        // 2) the thematic map menu has been changes
+        // 3) a new project is loaded into the map (case: weightChange = 0)
+    var selectedIndicatorLabel = $('#webGlThematicSelection option:selected').attr('class');
+
+    if (weightChange < parseInt(selectedIndicatorLabel || selectedIndicatorLabel !== '4')) {
```

Comment:
Please add a comment to clarify this line. It looks very cryptic to me. I don't understand `||` (a `&&` would look more intuitive, but we have to be sure about the logic). Also, please specify what does `weightChange` means and what kind of values it might take.